File: modules/AcroStack.AuditLogging/Dtos/AuditLogDto.cs
using System;
using System.Collections.Generic;

namespace AcroStack.AuditLogging;

public class AuditLogDto
{
    public Guid Id { get; set; }
    public Guid? UserId { get; set; }
    public string? UserName { get; set; }
    public Guid? TenantId { get; set; }
    public string? TenantName { get; set; }
    public string? HttpMethod { get; set; }
    public string? Url { get; set; }
    public int? HttpStatusCode { get; set; }
    public DateTime ExecutionTime { get; set; }
    public int ExecutionDuration { get; set; }
    public string? ClientIpAddress { get; set; }
    public string? ClientName { get; set; }
    public string? BrowserInfo { get; set; }
    public string? Exceptions { get; set; }
    public string? Comments { get; set; }
    public string? CorrelationId { get; set; }
    public List<EntityChangeDto> EntityChanges { get; set; } = new();
    public List<AuditLogActionDto> Actions { get; set; } = new();
}

File: react/src/api/models/acroStack/auditLogging/AuditLogDto.ts
/* oxlint-disable */

import type { AcroStackAuditLoggingAuditLogActionDto } from "./AuditLogActionDto";
import type { AcroStackAuditLoggingEntityChangeDto } from "./EntityChangeDto";

export type AcroStackAuditLoggingAuditLogDto = {
  /**
   * @description
   * Format: `uuid`
   * @type string | undefined
   */
  id?: string;
  /**
   * @description
   * Format: `uuid`
   * @type string | undefined
   */
  userId?: string | null;
  userName?: string | null;
  /**
   * @description
   * Format: `uuid`
   * @type string | undefined
   */
  tenantId?: string | null;
  tenantName?: string | null;
  httpMethod?: string | null;
  url?: string | null;
  /**
   * @description
   * Format: `int32`
   * @type integer | undefined
   */
  httpStatusCode?: number | null;
  /**
   * @description
   * Format: `date-time`
   * @type string | undefined
   */
  executionTime?: string;
  /**
   * @description
   * Format: `int32`
   * @type integer | undefined
   */
  executionDuration?: number;
  clientIpAddress?: string | null;
  clientName?: string | null;
  browserInfo?: string | null;
  exceptions?: string | null;
  comments?: string | null;
  correlationId?: string | null;
  entityChanges?: AcroStackAuditLoggingEntityChangeDto[] | null;
  actions?: AcroStackAuditLoggingAuditLogActionDto[] | null;
};
